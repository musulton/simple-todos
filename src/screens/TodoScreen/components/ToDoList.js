import React from 'react'
import {View} from 'react-native'
import ToDo from "./ToDo";

const ToDoList = ({todos,toggleComplete,deleteTodo,type}) => {
    const getVisibleTodos = (todos, type) => {
        switch (type) {
            case 'All':
                return todos
            case 'Complete':
                return todos.filter((t) => t.complete)
            case 'Active':
                return todos.filter((t) => !t.complete)
        } }
    let selectedTodos = getVisibleTodos(todos, type)
    let todoItems = selectedTodos.map((todo) => {
        return (
            <ToDo key={todo.todoIndex} todo={todo}  toggleComplete={toggleComplete} deleteTodo={deleteTodo}/>
        );
    })
    return (
        <View>
            {todoItems}
        </View>
    )
}
export default ToDoList;
