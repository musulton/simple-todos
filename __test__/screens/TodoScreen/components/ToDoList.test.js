import {fireEvent, render, screen} from "@testing-library/react-native";

import ToDoList from "../../../../src/screens/TodoScreen/components/ToDoList";
import {useSelector} from "react-redux";

jest
    .mock("react-redux", () => ({
        ...jest.requireActual("react-redux"),
        useSelector: jest.fn()
    }))
    .mock("../../../../src/shared/components/EmptyState", () => "EmptyState")

describe("ToDoList", () => {
    test("should return empty state correctly when data is empty", () => {
        const mockProps = {
            onDelete: jest.fn(),
            onCompleted: jest.fn()
        }

        render(<ToDoList {...mockProps} />)

        expect(screen.toJSON()).toMatchSnapshot()
    })

    test("should return correctly", () => {
        useSelector.mockImplementationOnce(() => [
            {title: 'title', complete: false, id: 1}
        ])
        useSelector.mockImplementationOnce(() => "All")

        const mockProps = {
            onDelete: jest.fn(),
            onCompleted: jest.fn()
        }

        render(<ToDoList {...mockProps} />)

        expect(screen.toJSON()).toMatchSnapshot()
    })
})
