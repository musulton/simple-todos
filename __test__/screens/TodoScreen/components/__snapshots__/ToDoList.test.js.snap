// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList should return correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={
    <EmptyState
      text="Data is still empty"
    />
  }
  contentContainerStyle={{}}
  data={
    [
      {
        "complete": false,
        "id": 1,
        "title": "title",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#ededed",
            "borderWidth": 1,
            "elevation": 3,
            "flexDirection": "row",
            "marginLeft": 20,
            "marginRight": 20,
            "paddingBottom": 7,
            "paddingLeft": 14,
            "paddingTop": 7,
            "shadowColor": "#000000",
            "shadowOffset": {
              "height": 3,
              "width": 3,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 3,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 16,
              },
              null,
            ]
          }
        >
          title
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <View
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "flex-end",
                "borderColor": "#ededed",
                "borderRadius": 4,
                "borderWidth": 1,
                "marginRight": 5,
                "padding": 7,
                "width": 50,
              }
            }
          >
            <Text />
          </View>
          <View
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "alignSelf": "flex-end",
                "borderColor": "#ededed",
                "borderRadius": 4,
                "borderWidth": 1,
                "marginRight": 5,
                "padding": 7,
                "width": 50,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`ToDoList should return empty state correctly when data is empty 1`] = `
<RCTScrollView
  ListEmptyComponent={
    <EmptyState
      text="Data is still empty"
    />
  }
  contentContainerStyle={
    {
      "flex": 1,
    }
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <EmptyState
      onLayout={[Function]}
      text="Data is still empty"
    />
  </View>
</RCTScrollView>
`;
